hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/HoodieSimpleBucketIndex.java

```diff
@@ -152,14 +153,18 @@ protected Function<HoodieRecord, Option<HoodieRecordLocation>> getIndexLocationF
 
   private class SimpleBucketIndexLocationFunction implements Function<HoodieRecord, Option<HoodieRecordLocation>> {
     private final Map<Integer, HoodieRecordLocation> bucketIdToFileIdMapping;
+    private final NumBucketsFunction numBucketsFunction;
 
     public SimpleBucketIndexLocationFunction(HoodieTable table, String partitionPath) {
       this.bucketIdToFileIdMapping = loadBucketIdToFileIdMappingForPartition(table, partitionPath);
+      HoodieWriteConfig writeConfig = table.getConfig();
+      this.numBucketsFunction = new NumBucketsFunction(writeConfig.getBucketIndexPartitionExpression(),
+          writeConfig.getBucketIndexPartitionRuleType(), writeConfig.getBucketIndexNumBuckets());
```

Comment:
can we add a factory method `NumBucketsFunction.fromWriteConfig` ?

Reply:
done